types/yup/yup-tests.ts

```diff
@@ -594,3 +599,41 @@ yup.object<MyInterface>({
         .required(),
     arrayField: yup.array(yup.string()).required()
 });
+
+const personSchema = yup.object({
+    firstName: yup.string(), // $ExpectType StringSchema<string>
+    email: yup
+        .string()
+        .nullable()
+        .email(),
+    birthDate: yup
+        .date()
+        .nullable()
+        .min(new Date(1900, 0, 1)),
```

Comment:
It would also be cool to add type assertions to this as well so that we're checking one of each for the nullable case!